### DISCUSSION TITLE
Run specific field/section validations? (Multi-Step Forms)

### DISCUSSION THREAD

**Discussion by DanjBethel** [filamentphp/filament#1846] [Help]
I'm wondering if there is a way to run a schema validation for a particular Section or grouping of fields.

I am thinking in terms of a multi-step form. Consider two sections, Patient Information and Payment Information. I want to validate the fields in payment information, and then uncollapse the payment information section if those validations passed.

Or maybe I don't know enough about filament to figure out a better way to do this.

How would you implement Multi-step forms using filament currently with the form builder?

**Comment by ryangjchandler** [comment]
@DanjBethel Sounds like you want to probably conditionally register fields / hide the hidden sections completely so they don't get validated.

**Reply by danharrin** [reply]
@ryangjchandler didn't you build a Wizard custom component before? I want to add something like that to the core. 1 component container per step, validate the container when switching steps.

**Reply by DanjBethel** [reply]
I imagine this would be a great addition. Sometimes you may not want to show the entire form, as with the "Tabs" feature, but may also want to validate a particular grouping of fields at a time. These implementations don't have to be specifically tied to "Multi-Step Forms", but together you could show in the documentation how one can be created using both techniques.

Or a Steps>Step Component can be a thing.

```
Steps::make()->scheme([
  Step::make()->schema([
    // Inputs and their validations
  ])
  // Gets fired when section is stepped through
  ->onValidated()
  // Gets error bag for this step?
  ->onError()
])

```

Just freestyling what a possible PR could look like.

**Reply by madsem** [reply]
This would actually be very useful for external data as well. My use case currently: add external accounts via API, each account can have sub-accounts, which could be selected in a multi step form.

1. Enter connection data
2. Make API call and display available Accounts 
3. Save selections

**Comment by danharrin** [comment]
We've added a wizard component now!

https://filamentphp.com/docs/2.x/forms/layout#wizard
https://filamentphp.com/docs/2.x/admin/resources/creating-records#wizards
https://filamentphp.com/docs/2.x/tables/actions#wizards
